en/protocol/mission.md

```diff
@@ -29,9 +53,39 @@ sequenceDiagram;
 {% endmermaid %}
 
 
+In more detail, the sequence of operations is:
+1. GCS (client) sends [MISSION_COUNT](../messages/common.md#MISSION_COUNT) including the number of mission items to be uploaded (`count`)
+   - A [timeout](#timeout) must be started for the GCS to wait on the response from Drone (`MISSION_REQUEST`) .
+1. Drone (server) receives the message, and prepares to upload mission items.
+1. Drone responds with [MISSION_REQUEST](../messages/common.md#MISSION_REQUEST) requesting the mission number to upload (`seq`)
+1. GCS receives `MISSION_REQUEST` and responds with the requested mission item in a [MISSION_ITEM](../messages/common.md#MISSION_ITEM) message.
+1. Drone and GCS repeat the `MISSION_REQUEST`/`MISSION_ITEM` cycle until all items are uploaded.
+1. For the last mission item, the drone responds with [MISSION_ACK](../messages/common.md#MISSION_ACK) with the result of the operation result: `type` ([MAV_MISSION_RESULT](../messages/common.md#MAV_MISSION_RESULT)):
+   - On success, `type` must be set to [MAV_MISSION_ACCEPTED](../messages/common.md#MAV_MISSION_ACCEPTED)
+   - On failure, `type` must set to [MAV_MISSION_ERROR](../messages/common.md#MAV_MISSION_ERROR) or some other error code. 
+1. GCS receives `MISSION_ACK`:
+   - If `MAV_MISSION_ACCEPTED` the operation is complete.
+   - If an error, the transaction fails but may be retried. <!-- not clear here -->
```

Comment:
I am not 100% sure of the handling here if the transaction fails. It appears to be that there is cleanup and an error message.